test/libsolidity/syntaxTests/literals/ternary_operator_return_type.sol

```diff
@@ -0,0 +1,19 @@
+contract TestTernary
+{
+    mapping(uint8 => bool) flags;
+
+    function g() public returns (uint256 result)
+    {
+        flags[uint8(1)] = true;
+        flags[uint8(2)] = true;
+        flags[uint8(3)] = true;
+
+        result = (flags[0] ? uint256(100) : 0) +
+                    (flags[1] ? 110 : 0) +
+                    (flags[2] ? 125 : 0) +
+                    (flags[3] ? 150 : 0);
```

Comment:
I think we should also have semantic test to verify the current unintuitive behavior. I.e. that these actually overflow:
```solidity
        uint16 a = (t ? 63 : 255) + (f ? 63 : 255);
```
```solidity
        uin16 a = (true ? 63 : 255) + (false ? 63 : 255);
```